# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/painting/RetainedDisplayListBuilder.cpp
# Commit: c09a51622e98
# Full Hash: c09a51622e988e81077c75e3aaa5a49df66284b3
# Author: Doug Thayer <dothayer@mozilla.com>
# Date: 2019-03-22 16:44:08
# Regressor Bug: 1441308
# File Overlap Count: 2
# Description:
#   Bug 1441308 - Core renderroot splitting changes r=kats,sotaro
#   
#   This is a large patch that contains all of the core changes for
#   renderroot splitting.
#   
# ==============================================================================

diff -r 737807563dd5 -r c09a51622e98 layout/painting/RetainedDisplayListBuilder.cpp
--- a/layout/painting/RetainedDisplayListBuilder.cpp	Thu Mar 21 23:14:26 2019 +0000
+++ b/layout/painting/RetainedDisplayListBuilder.cpp	Fri Mar 22 03:15:14 2019 +0000
@@ -364,7 +364,7 @@
           if (mBuilder->MergeDisplayLists(
                   aNewItem->GetChildren(), oldItem->GetChildren(),
                   destItem->GetChildren(), containerASRForChildren, aNewItem)) {
-            destItem->InvalidateCachedChildInfo();
+            destItem->InvalidateCachedChildInfo(mBuilder->Builder());
             mResultIsModified = true;
           }
           UpdateASR(destItem, containerASRForChildren);
@@ -504,6 +504,7 @@
       Span<const MergedListIndex> aDirectPredecessors,
       const Maybe<MergedListIndex>& aExtraDirectPredecessor) {
     UpdateContainerASR(aItem);
+    aItem->NotifyUsed(mBuilder->Builder());
 
 #ifdef MOZ_DIAGNOSTIC_ASSERT_ENABLED
     nsIFrame::DisplayItemArray* items =
@@ -542,7 +543,7 @@
         if (mBuilder->MergeDisplayLists(&empty, item->GetChildren(),
                                         item->GetChildren(),
                                         containerASRForChildren, item)) {
-          item->InvalidateCachedChildInfo();
+          item->InvalidateCachedChildInfo(mBuilder->Builder());
           mResultIsModified = true;
         }
         UpdateASR(item, containerASRForChildren);